packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -178,7 +189,44 @@ async function processBlogSourceFile(
     } else if (parsedBlogFileName.date) {
       return parsedBlogFileName.date;
     }
-    // Fallback to file create time
+
+    if (!shell.test('-f', blogSourceAbsolute)) {
```

Comment:
please extract all this low-level git logic in a separate well-named and tested function like `getGitCreationDate(source)` or something.

I feel like we could reuse this in other places so maybe add `gitUtils.ts` in `@docusaurus/utils` ?

Reply:
Yes, this is very related to [lastUpdate.ts](https://github.com/facebook/docusaurus/pull/6592). I do agree with you, it would be better if it could be reused.

If I add this to @docusaurus/utils, I will have to add `shelljs` as its dependency as well. Some other packages relying on utils which do not require this function may end up depending on shelljs without need. Is it okay?

Reply:
core depends on shelljs so I think it's fine

Reply:
@felipecrs Literally every Docusaurus site has this as a dependency because the core uses this. Also, people seldom worry about transitive dependencies.

Reply:
Alright. I'll work on it. :)